src/extendables/User/Minion.ts

```diff
@@ -587,26 +588,46 @@ export default class extends Extendable {
 	}
 
 	public async getKCByName(this: KlasaUser, kcName: string) {
+		const specialKCs = [
+			[['superior', 'superiors', 'superior slayer monster'], UserSettings.Slayer.SuperiorCount],
+			[['tithefarm', 'tithe'], UserSettings.Stats.TitheFarmsCompleted]
+		];
+
 		const mon = [
 			...killableMonsters,
 			NightmareMonster,
 			{ name: 'Zalcano', aliases: ['zalcano'], id: ZALCANO_ID },
-			{ name: 'TzTokJad', aliases: ['jad', 'fightcaves'], id: TzTokJad.id }
+			{ name: 'TzTokJad', aliases: ['jad', 'fightcaves'], id: TzTokJad.id },
+			{ name: 'Hespori', aliases: ['hespori'], id: HESPORI_ID }
 		].find(mon => stringMatches(mon.name, kcName) || mon.aliases.some(alias => stringMatches(alias, kcName)));
 		const minigame = Minigames.find(game => stringMatches(game.name, kcName));
 		const creature = Creatures.find(c => c.aliases.some(alias => stringMatches(alias, kcName)));
 
-		if (!mon && !minigame && !creature) {
+		let specialName = '';
+		let specialSetting = null;
+
+		outerloop: for (const [names, setting] of specialKCs) {
+			for (const name of names) {
+				if (stringMatches(name, kcName)) {
+					[specialName, specialSetting] = [name, setting];
+					break outerloop;
+				}
+			}
+		}
+
+		if (!mon && !minigame && !creature && !specialSetting) {
```

Comment:
I really dont think a nested loop is needed here, either do: a `.find()` or `.some()` inside the loop:

something like this
```ts
const specialKC = specialKCs.find(s => s[0].some(alias => stringMatches(alias, name))
```